lib/retry.ts

```diff
@@ -122,6 +142,14 @@ export async function retry<T>(config: RetryConfig<T>): Promise<T> {
       if (!err.translated) {
         err = translate(err);
       }
+
+      if (!err.retryable && err.name === "ServiceCommunicationError" && config.connectionHost) {
+        const isConnected = await checkNetworkConnection(config.connectionHost);
```

Comment:
We may need to pass in port number too, refer to main comment for more context.

Reply:
I don't think `dns.resolve` needs the port. It sure doesnt support accepting it as a parameter.

Reply:
I think we don't need to pass a port for probing DNS. DNS lookups take a domain name only. DNS will use UDP over port 53 by default and users machines should be configured to use DNS over this port else using internet resources would be very tough.